### PR TITLE
notification: circ policies new parameters

### PR DESCRIPTION
**Pull request by BadrAly**
* Adds notification parameters to circulation policy.
* number_of_days_before_due_date: needed for the due_soon notification.
* Rnumber_of_days_after_due_date: needed for the over_due first_reminder notification.

Co-Authored-by: Aly Badr <aly.badr@rero.ch>
Co-Authored-by: Peter Weber <peter.weber@rero.ch>

# To test:
* `./scripts/bootstrap`
* `./scripts/setup`
* edit one of the policies at `http://localhost:5000/admin/records/circ_policies` and check out the new parameters.

### CODE REVIEW

File: rero_ils/modules/circ_policies/jsonschemas/circ_policies/circ_policy-v0.0.1.json [hunk 1]
```diff
@@ -65,6 +65,20 @@
       "type": "boolean",
       "default": true
     },
+    "number_of_days_before_due_date": {
+      "title": "Number of days before due data.",
+      "description": "Number of days before due date for sedning the availability notification.",
```
Comment (Garfield-fr) [inline rero_ils/modules/circ_policies/jsonschemas/circ_policies/circ_policy-v0.0.1.json]: Typo on word sedning. Change with sending.

File: rero_ils/modules/circ_policies/jsonschemas/circ_policies/circ_policy-v0.0.1.json [hunk 2]
```diff
@@ -65,6 +65,20 @@
...
+      "type": "integer",
+      "default": 5,
+      "minimum": 1
+    },
+    "number_of_days_after_due_date": {
+      "title": "Number of days after due data.",
+      "description": "Number of days after due date for sedning the first overdue notification.",
```
Comment (Garfield-fr) [inline rero_ils/modules/circ_policies/jsonschemas/circ_policies/circ_policy-v0.0.1.json]: Typo on word sedning. Change with sending.

**Comment by jma** [comment]
please change the commit message with the new guideline

File: ui/src/app/records/custom-editor/circulation-settings/circulation-policy-form.service.ts [hunk 3]
```diff
@@ -75,9 +81,18 @@ export class CirculationPolicyFormService {
           Validators.required,
           Validators.min(0)
         ]);
+        daysAfterControl.setValidators([
+          Validators.required,
+          Validators.min(0)
+        ]);
+        daysBeforeControl.setValidators([
+          Validators.required,
+          Validators.min(0)
```
Comment (jma) [inline ui/src/app/records/custom-editor/circulation-settings/circulation-policy-form.service.ts]: the json schema say: 1!

File: ui/src/app/records/custom-editor/circulation-settings/circulation-policy-form.service.ts [hunk 4]
```diff
@@ -75,9 +81,18 @@ export class CirculationPolicyFormService {
           Validators.required,
           Validators.min(0)
         ]);
+        daysAfterControl.setValidators([
+          Validators.required,
+          Validators.min(0)
```
Comment (jma) [inline ui/src/app/records/custom-editor/circulation-settings/circulation-policy-form.service.ts]: the json schema say: 1!

File: ui/src/app/records/custom-editor/circulation-settings/circulation-policy/circulation-policy.component.html [hunk 5]
```diff
@@ -70,11 +70,53 @@ <h1 translate>Circulation Policy</h1>
...
+        class="form-control"
+        id="daysAfterControl"
+        formControlName="number_of_days_after_due_date"
+      >
+      <div *ngIf="number_of_days_after_due_date.invalid && (number_of_days_after_due_date.dirty || number_of_days_after_due_date.touched)" class="text-danger pt-1">
+        <div *ngIf="number_of_days_after_due_date.errors.required" translate>
+          First reminder days is required.
+        </div>
+        <div *ngIf="number_of_days_after_due_date.errors.min" translate>
+          First reminder days must be greater than 0.
```
Comment (jma) [inline ui/src/app/records/custom-editor/circulation-settings/circulation-policy/circulation-policy.component.html]: or 1?

**Comment by jma** [comment]
I did not see any modification in the ES mapping for the two new fields in circulation policies

File: ui/src/app/records/custom-editor/circulation-settings/circulation-policy-form.service.ts (hunk 3 above)
Comment (BadrAly) [reply ui/src/app/records/custom-editor/circulation-settings/circulation-policy-form.service.ts]: fixed

File: ui/src/app/records/custom-editor/circulation-settings/circulation-policy-form.service.ts (hunk 4 above)
Comment (BadrAly) [reply ui/src/app/records/custom-editor/circulation-settings/circulation-policy-form.service.ts]: fixed

File: ui/src/app/records/custom-editor/circulation-settings/circulation-policy/circulation-policy.component.html (hunk 5 above)
Comment (BadrAly) [reply ui/src/app/records/custom-editor/circulation-settings/circulation-policy/circulation-policy.component.html]: greater than 0 is correct

**Comment by BadrAly** [comment]
> I did not see any modification in the ES mapping for the two new fields in circulation policies

Mappings are up to date .

**Comment by BadrAly** [comment]
> please change the commit message with the new guideline

done

**Pull request merged**